### ISSUE TITLE
Confusing UX for preview panel

### ISSUE THREAD

**Issue by rugk** [keepassxreboot/keepassxc#1730]
## Currently
* I click an entry. -> Currently the new preview panel is opened.
* I click the "close icon" at the top right of it. -> It is hidden, unless you select yet another item.
* when I disable the option to show the preview panel in the options -> It is permanently hidden, _but not for the current item, when shown_:

## What's bad?

First, I don't see any use case / usefulness for the second thing. When I click a close icon to hide a preview (whether it is a mail preview or something else) I want it to be hidden, e.g. I want more space for my passwords.
When it is then shown again when I select yet another item, that is not useful, as I just want to hide it again.
But as someone mkay find it useful, I think we do not need to remove that entire thing.

You could, however, just combine it with the selection of the password entry. I.e. when you click the X icon, also unselect the entry. This would make it clear that with a reselection, the preview is shown again.

Additionally the option to show/hide the panel permanently is hidden too deep in the settings. And of course, it should apply immediately and not "on the next click on an entry".

## How it should be
* I click an entry. -> Currently the new preview panel is opened.
* I click the "close icon" at the top right of it. -> It is hidden and deselected. OR hide permanently (disable preview panel)
* Move preview panel option into menu, not options, so it can be reached faster (as users might toggle it on or off when they need it for a second or so)
* when I disable the preview option (in the menu) -> It is permanently hidden, and, if opened, the current preview is hidden.

**Comment by TheZ3ro** [comment]
Check out the spacer on top of the preview panel, you can drag it to the bottom to hide it. This way when selecting a new entry the panel wont open again.

The X button is just a temporary way to hide the panel when you are looking into the entry list but you still want it again when clicking on an element.
The setting will permanently hide the panel no matter what.

**Comment by droidmonkey** [comment]
I think the :x: should be replaced with a down arrow like: ![down_arrow](https://user-images.githubusercontent.com/2809491/37556566-7dbcb402-29ce-11e8-8cc7-b94fa7cbbd0f.png)

**Comment by TheZ3ro** [comment]
Seems fine to me

**Comment by rugk** [comment]
@droidmonkey Even the :arrow_down: button implies the same. Unless I click an "arrow up :arrow_up: " button again, the panel must stay hidden. So replacing a button is no solution.

And BTW, I stil see no valid use case for a "temporary hiding" the panel. Why should I bide it temporary just to let it come back when I click "anywhere"? When I want to see throuugh the list I can just see it like that. Or, I have an :arrow_down: button, click it to see the list, click the :arrow_up: button to see the details again.

**Comment by rugk** [comment]
> Check out the spacer on top of the preview panel, you can drag it to the bottom to hide it.

And when hidden I really have to drag it back to see it? No, that's not so nice…
Especially as you actually have a setting in the options for hiding that panel. One then does not even know how the panel may be hidden and how to restore it.

**Comment by droidmonkey** [comment]
Ideally when you hit the down arrow it woukd collapse to a minimum height that shows a title bar "Entry Details" and an up arrow.

**Comment by rugk** [comment]
Indeed that's what I'd expect/imagine for a down arrow. That there is an up arrow, where I can show it again.

**Comment by TheZ3ro** [comment]
> And BTW, I stil see no valid use case for a "temporary hiding" the panel.

I've told you above, some users want to temporary hide the panel to see through the "complete" entry list. The panel opened is hiding 30% of it.
But then if I click on an entry I still want to see its details.

> And when hidden I really have to drag it back to see it? No, that's not so nice…

That's how spacers works

Anyway, we already discussed about a behavior like the Up/Down arrow and I think it's a pretty good compromise.

*Like all UX issues, since every user use KeePassXC differently, "fixing" a thing for one "break" the flow for another*

**Comment by droidmonkey** [comment]
We should use this for the details panel, I think it would greatly enhance our UX.

https://github.com/Elypson/qt-collapsible-section

**Comment by rugk** [comment]
Yeah, the sample there looks good, also with animation. :smile:

**Comment by kneitinger** [comment]
I think we should get this done before v2.4.0. I'm noticing some odd interplay of the group preview panel vs. the entry preview panel when one (or both) of them is closed, and a different group is clicked, or an entry is edited.  When that condition is met, the preview panel only displays either the last clicked group, or the last edited entry (which ever happened most recently).

**Comment by phoerious** [comment]
It's the same widget.

**Comment by kneitinger** [comment]
Right, good point.  Group and Entry each have a their own sub-QWidgets (pageEntry and pageGroup) that each have their own independent close buttons. I would _guess_ that there is logic that is calling `show()` and/or `hide()` on those individually which is introducing some inconsistencies in the preview widget that cause it to sometimes behave as though it is two separate widgets.

**Comment by droidmonkey** [comment]
This was improved for 2.7.10
